test/python/circuit/test_random_circuit.py

```diff
@@ -160,3 +163,196 @@ def test_random_circuit_with_zero_distribution(self):
         self.assertEqual(gate_type_counter[1], 0.0)
         self.assertEqual(gate_type_counter[2], 0.0)
+
+
+class TestRandomCircuitFromGraph(QiskitTestCase):
+    """Testing random_circuit_from_graph from
+    qiskit.circuit.random.utils.py"""
+
+    def setUp(self):
+        super().setUp()
+        n_q = 10
+        pydi_graph = rx.PyDiGraph()
+        pydi_graph.add_nodes_from(range(n_q))
+        # Some arbitrary coupling map
+        cp_map = [
+            (0, 2, None),
+            (1, 3, None),
+            (2, 4, None),
+            (3, 4, None),
+            (5, 7, None),
+            (4, 7, None),
+            (7, 9, None),
+            (5, 8, None),
+            (6, 9, None),
+        ]
+        pydi_graph.add_edges_from(cp_map)
+        self.interaction_graph = (pydi_graph, None, None, None)
+
+    def test_simple_random(self):
+        """Test creating a simple random circuit."""
+        circ = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph, min_2q_gate_per_edge=1
+        )
+        self.assertIsInstance(circ, QuantumCircuit)
+        self.assertEqual(circ.width(), 10)
+
+    def test_min_times_qubit_pair_usage(self):
+        """the `min_2q_gate_per_edge` parameter specifies how often each qubit-pair must at
+        least be used in a two-qubit gate before the circuit is returned"""
+
+        freq = 1
+        qc = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph, min_2q_gate_per_edge=freq, seed=0
+        )
+        dag = circuit_to_dag(qc)
+        count_register = defaultdict(int)
+
+        for wire in dag.wires:
+            for node in dag.nodes_on_wire(wire, only_ops=True):
+                if node.op.name == "measure" or node.op.num_qubits < 2:
+                    continue
+                key = node.sort_key
+                control, target = key.split(",")
+                count_register[(control, target)] += 1
+
+        for occurence in count_register.values():
+            self.assertLessEqual(freq, occurence)
+
+    def test_random_measure(self):
+        """Test random circuit with final measurement."""
+        qc = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph, min_2q_gate_per_edge=1, measure=True
+        )
+        self.assertIn("measure", qc.count_ops())
+
+    def test_random_circuit_conditional_reset(self):
+        """Test generating random circuits with conditional and reset."""
+        # Presence of 'reset' in the circuit is probabilistic, at seed 0 reset exists in circuit.
+        qc = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph,
+            min_2q_gate_per_edge=1,
+            conditional=True,
+            reset=True,
+            seed=486,  # Do not change the seed or the args.
+            insert_1q_oper=True,
+        )
+        self.assertIn("reset", qc.count_ops())
```

Comment:
You could also check for conditional gates here

Reply:
strategy was to focus on testing only one feature in one test, but I will add code to check for conditionals here only.

Reply:
Done :)